bin/wash.js

```diff
@@ -23,6 +28,120 @@ if ('setRawMode' in process.stdin) {
   process.stdin.setRawMode(true);
 }
 
+var WebSocketServer = require('ws').Server;
+
+var WebSocketFs = function(cx, port, fileSystemName) {
+  this.cx_ = cx;
+  this.port_ = port;
+  this.fileSystemName_ = fileSystemName;
+  this.wss_ = new WebSocketServer({ port: port });
+  this.completer_ = new Completer();
+  this.cx_.onClose.addListener(function() {
+    this.println('closing server');
+    this.wss_.close();
+    this.completer_.resolve();
+  }.bind(this));
+};
+
+WebSocketFs.prototype.println = function(msg) {
+  this.cx_.stdout.write(msg + '\n');
+};
+
+WebSocketFs.prototype.run = function() {
+  this.wss_.on('connection', function (ws) {
+    try {
+     this.openFileSystem(ws);
+     this.println('WebSocket file system connection accepted');
+    }
+    catch(error) {
+      this.println('WebSocket file system connection error: ' + error.message);
+      console.log(error);
+      ws.close();
+    }
+  }.bind(this));
+
+  this.println('WebSocket server for file system "' + this.fileSystemName_ +
+      '" running on port ' + this.port_);
+  this.println('Waiting for connections, press Ctrl-C to terminate.');
+  return this.completer_.promise;
+};
+
+WebSocketFs.prototype.openFileSystem = function(webSocket) {
+  var fileSystem = null;
+  this.cx_.fileSystemManager.getFileSystems().forEach(function(fs) {
+    if (fs.rootPath.root === this.fileSystemName_) {
+      fileSystem = fs;
```

Comment:
Good point...Well, `some` does not work because it returns a boolean, and `filter` returns an array. What I really need is `FirstOrDefault` (https://msdn.microsoft.com/en-us/library/vstudio/bb549039(v=vs.100).aspx)  :)

I'll change to use filter then...